# Model testing and validation

# Imports

In [1]:
import joblib
import scipy.sparse as sp
import pandas as pd
from sklearn.metrics import classification_report

## Import models

In [2]:
tfidf = joblib.load('csv/tfidf_dump.pkl')
model = joblib.load('csv/sentiment_model.pkl')

X_test  = sp.load_npz('csv/X_test.npz')
y_test  = pd.read_csv('csv/y_test.csv')['Sentiment']

# Testing

In [3]:
y_pred = model.predict(X_test)
print(classification_report(y_test, y_pred, 
      target_names=['Neutral', 'Negative', 'Positive']))

              precision    recall  f1-score   support

     Neutral       0.79      0.85      0.82      6059
    Negative       0.83      0.80      0.82      4383
    Positive       0.81      0.75      0.78      4032

    accuracy                           0.81     14474
   macro avg       0.81      0.80      0.81     14474
weighted avg       0.81      0.81      0.81     14474



todo: analyse the spat out data

Under initial testing (with my first iteration, just the twitter dataset), my accuracy was at 81% accuracy

|   | precision | recall | f1-score | support |
| - | --------- | ------ | -------- | ------- |
| Neutral | 0.79 | 0.85 | 0.82 | 6059 |
| Negative | 0.83 | 0.80 | 0.82 | 4383 |
| Positive | 0.81 | 0.75 | 0.78 | 4032 |
| |
| Accuracy | | | 0.81 | 14474 |
| Macro Average | 0.81 | 0.80 | 0.81 | 14474 |
| Weighted Average | 0.81 | 0.81 | 0.81 | 14474 |